metldata/builtin_transformations/delete_slots/main.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A tranformation for deleting slots from classes in a metadata model."""
+
+from metldata.builtin_transformations.delete_slots.assumptions import (
+    check_model_class_slots,
+)
+from metldata.builtin_transformations.delete_slots.config import SlotDeletionConfig
+from metldata.builtin_transformations.delete_slots.metadata_transform import (
+    delete_class_slots,
+)
+from metldata.builtin_transformations.delete_slots.model_transform import (
+    delete_class_slots_from_model,
+)
+from metldata.model_utils.anchors import get_anchors_points_by_target
+from metldata.model_utils.assumptions import check_anchor_points
+from metldata.model_utils.essentials import MetadataModel
+from metldata.transform.base import Json, MetadataTransformer, TransformationDefinition
+
+
+def check_model_assumptions(model: MetadataModel, config: SlotDeletionConfig):
+    """Check that the classes and slots specified in the config exist in the model."""
+
+    check_model_class_slots(model=model, class_slots=config.slots_to_delete)
+    check_anchor_points(model=model, classes=list(config.slots_to_delete.keys()))
```

Comment:
```suggestion
    check_anchor_points(model=model, classes=list(config.slots_to_delete))
```

The dict itself already iterates over the keys, so `keys()` is usually not needed.

If `check_anchor_points` would accept any `Sequence` instead of a `list`, then the `list` call would not be needed either.